mlir-sycl/lib/Dialect/SYCL/Utils/Utils.cpp

```diff
@@ -16,20 +16,23 @@ using namespace mlir;
 using namespace mlir::sycl;
 
 SYCLIDGetOp sycl::createSYCLIDGetOp(TypedValue<MemRefType> id, unsigned index,
-                                    OpBuilder builder, Location loc) {
+                                    bool memrefTy, OpBuilder builder,
```

Comment:
I rather don't have to pass a bool and instead pass to this utility functions the result type of the operation they need to create. So:
```
SYCLIDGetOp sycl::createSYCLIDGetOp(TypedValue<MemRefType> id, unsigned index, Type resTy,                                     OpBuilder builder, Location log);